### ISSUE TITLE
Use -Wno-ignored-attributes

### ISSUE THREAD

**Issue by tvatter** [RcppCore/RcppEigen#59]
Hi,

This is not really an issue with RcppEigen, but I am not sure where would be the appropriate place to discuss the following.

I'm working on a mac, and, since update 0.3.3.4.0, I get hundreds of the `ignoring attributes on template argument` warnings whenever I compile a package relying on (Rcpp)Eigen.

This is notably the case with [a package that I develop](https://cran.r-project.org/web/packages/rvinecopulib/index.html). In this case, there are so many warnings that I have to wait around for 5-10mn every time I try to compile (versus <1mn using `-Wno-ignored-attributes`)

Hence, we added `PKG_CXXFLAGS = -Wno-ignored-attributes` to [our Makevars in the dev branch](https://github.com/vinecopulib/rvinecopulib/blob/dev/src/Makevars), and we are now ready for a new release, but, when running the cran checks, we get:

```
checking compilation flags in Makevars ... WARNING
Non-portable flags in variable 'PKG_CXXFLAGS':
  -Wno-ignored-attributes

checking compilation flags used ... WARNING
Compilation used the following non-portable flag(s):
  ‘-Wno-ignored-attributes’
including flag(s) suppressing warnings
```

What is the appropriate way to deal with this issue?

**Comment by eddelbuettel** [comment]
You can only address it via your local compiler options as I recently explained somewhere (possibly r-package-devel list).

**Comment by tvatter** [comment]
My apologies, I looked around and didn't see such a discussion. I can obviously add the option to my local  `~/.R/Makevars`, but it means that other mac users trying to compile the package will have to wait forever for its compilation. Thus, I wonder if anyone had found a better solution...

**Comment by tvatter** [comment]
For instance, doing something like changing the build options for mac vs linux as in the link below?

https://stackoverflow.com/questions/32851240/change-build-options-for-mac-vs-linux-in-r-package

**Comment by eddelbuettel** [comment]
You can muck around with configure and add them dynamically despite the fact that CRAN tells you not to. You may still get caught. There is no silver bullet.

**Comment by eddelbuettel** [comment]
Yes, as you said over there: "unfortunate" but the best we can do. :pensive:
